clients/wallet/src/task.rs

```diff
@@ -0,0 +1,206 @@
+use crate::Ledger;
+use tokio::sync::oneshot::{channel, error::TryRecvError, Receiver, Sender};
+
+pub const LEDGER_TASK_MAX_RETRIES: u8 = 3;
+/// Determines the status of the task of processing a transaction
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum LedgerTaskStatus {
+	NotStarted,
+	/// Process Done and it was a success
+	ProcessSuccess,
+	/// The error is acceptable, and is best to just retry again
+	RecoverableError,
+	/// Something happened when
+	Error(String),
+}
+
+pub struct LedgerTask {
+	ledger: Ledger,
```

Comment:
I think this implementation is so general that in theory, we could generify it so that it is not specific to just 'ledgers' but to anything.  You implemented some struct that has some kind of ID (the `ledger` field) and is able to hold info about its current status, and can retry some number of times. 
If we generify the ID passing we could also use this in other contexts. Counter-argument would be that we over-engineer this while we don't need to, but since it's little extra work I would propose we add those changes too. WDYT @b-yap ?

Reply:
In any case, I would like to rename this. The name would be different if we generify, however.

Reply:
I just changed it to `SlotTask` because there's no alternate name provided. Do you have any suggestions (even just a temp one would be fine  :v:)?

And yes, creating a generic out of this struct  did cross my mind, but I cannot find yet on applying it elsewhere.
Should we create a ticket and explore the "possibilities" from there?  :clap: :grinning: